File: dotnet/ExtentReports/LogCounts.cs
// ***********************************************************************
// Copyright (c) 2015, Anshoo Arora (Relevant Codes). All rights reserved.
//
// Copyrights licensed under the New BSD License.
//
// See the accompanying LICENSE file for terms.
// ***********************************************************************

namespace RelevantCodes.ExtentReports
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;

    using Model;

    internal class LogCounts
    {        
        public LogCounts GetLogCounts(Test test) {
            Pass += test.Logs.Where(l => l.LogStatus == LogStatus.Pass).Count();
            Fail += test.Logs.Where(l => l.LogStatus == LogStatus.Fail).Count();
            Fatal += test.Logs.Where(l => l.LogStatus == LogStatus.Fatal).Count();
            Error += test.Logs.Where(l => l.LogStatus == LogStatus.Error).Count();
            Warning += test.Logs.Where(l => l.LogStatus == LogStatus.Warning).Count();
            Info += test.Logs.Where(l => l.LogStatus == LogStatus.Info).Count();
            Skip += test.Logs.Where(l => l.LogStatus == LogStatus.Skip).Count();
            Unknown += test.Logs.Where(l => l.LogStatus == LogStatus.Unknown).Count();

            // recursively count for all child tests
            test.NodeList.ForEach(n => GetLogCounts(n));
    		
    		return this;
    	}

        public int Pass
        {
            get;
            private set;
        }

        public int Fail
        {
            get;
            private set;
        }

        public int Fatal
        {
            get;
            private set;
        }

        public int Error
        {
            get;
            private set;
        }

        public int Warning
        {
            get;
            private set;
        }

        public int Info
        {
            get;
            private set;
        }

        public int Skip
        {
            get;
            private set;
        }

        public int Unknown
        {
            get;
            private set;
        }

        public LogCounts() { }
    }
}

File: java/extentreports/src/main/java/com/relevantcodes/extentreports/LogCounts.java
package com.relevantcodes.extentreports;

import java.util.HashMap;

import com.relevantcodes.extentreports.model.Test;

public class LogCounts {
    private int pass = 0;
    private int fail = 0;
    private int fatal = 0;
    private int error = 0;
    private int warning = 0;
    private int info = 0;
    private int skip = 0;
    private int unknown = 0;
    
    // counts each type of log for the test
    public HashMap<LogStatus, Integer> getLogCounts(Test test) {
    	for (int ix = 0; ix < test.getLog().size(); ix++) {
            if (test.getLog().get(ix).getLogStatus() == LogStatus.PASS)
                pass++; 
            else if (test.getLog().get(ix).getLogStatus() == LogStatus.FAIL)
            	fail++;
            else if (test.getLog().get(ix).getLogStatus() == LogStatus.FATAL)
            	fatal++;
            else if (test.getLog().get(ix).getLogStatus() == LogStatus.ERROR)
            	error++;
            else if (test.getLog().get(ix).getLogStatus() == LogStatus.WARNING)
            	warning++;
            else if (test.getLog().get(ix).getLogStatus() == LogStatus.INFO)
            	info++;
            else if (test.getLog().get(ix).getLogStatus() == LogStatus.SKIP)
            	skip++;
            else if (test.getLog().get(ix).getLogStatus() == LogStatus.UNKNOWN)
            	unknown++;
        }

        // recursively count status events
        for (Test node : test.getNodeList()) {
            getLogCounts(node);
        }
        
        HashMap<LogStatus, Integer> logCounts = new HashMap<LogStatus, Integer>();
    	
        logCounts.put(LogStatus.PASS, pass);
    	logCounts.put(LogStatus.FAIL, fail);
    	logCounts.put(LogStatus.FATAL, fatal);
    	logCounts.put(LogStatus.ERROR, error);
    	logCounts.put(LogStatus.WARNING, warning);
    	logCounts.put(LogStatus.INFO, info);
    	logCounts.put(LogStatus.SKIP, skip);
    	logCounts.put(LogStatus.UNKNOWN, unknown);
    	
    	return logCounts;
    }

    public LogCounts() { }
}